sdk/core/core-http/lib/util/xml.ts

```diff
@@ -34,4 +33,171 @@ export function parseXML(str: string): Promise<any> {
       });
     }
   });
+  return result;
+}
+
+export async function deserializeAtomXmlToJson(body: string): Promise<any> {
+  const parser = new xml2js.Parser(_getDefaultSettingsForAtomXmlOperations());
+  const result = await new Promise((resolve, reject) => {
+    parser.parseString(_removeBOM(body.toString()), function(err: any, parsedBody: any) {
+      if (err) {
+        reject(err);
+      } else {
+        resolve(parsedBody);
+      }
+    });
+  });
+  return result;
+}
+
+/**
+ * @ignore
+ * @param {object} content The content payload as it is to be serialized. It should include any root node(s).
+ */
+export function serializeJsonToAtomXml(content: any): string {
+  content[Constants.XML_METADATA_MARKER] = { type: "application/xml" };
+
+  let doc = xmlbuilder.create();
+
+  doc = doc
+    .begin("entry", { version: "1.0", encoding: "utf-8", standalone: "yes" })
+    .att("xmlns", "http://www.w3.org/2005/Atom");
+
+  doc = doc.ele("updated", new Date().toISOString()).up();
+
+  doc = _writeElementValue(doc, "content", content);
+  return doc.doc().toString();
+}
+
+/**
+ * @ignore
+ * Gets the default xml2js settings applicable for Atom based XML operations.
+ */
+function _getDefaultSettingsForAtomXmlOperations(): any {
+  const xml2jsSettings = xml2js.defaults["0.2"];
+  xml2jsSettings.normalize = false;
+  xml2jsSettings.trim = false;
+  xml2jsSettings.attrkey = "$";
+  xml2jsSettings.charkey = "_";
+  xml2jsSettings.explicitCharkey = false;
+  xml2jsSettings.explicitArray = false;
+  xml2jsSettings.ignoreAttrs = true;
+  return xml2jsSettings;
+}
+
+/**
+ * @ignore
+ * Gets the default settings applicable for general XML operations.
+ */
+function _getDefaultSettings(): any {
+  const xml2jsSettings = xml2js.defaults["0.2"];
+  xml2jsSettings.explicitArray = false;
+  xml2jsSettings.ignoreAttrs = false;
+  xml2jsSettings.explicitCharkey = false;
+  xml2jsSettings.explicitRoot = false;
+  xml2jsSettings.normalize = true;
+  return xml2jsSettings;
+}
+
+/**
+ * @ignore
+ * @param str Helper utility to clean up unintended characters that get appended by OS.
+ */
+function _removeBOM(str: any) {
+  if (str.charCodeAt(0) === 0xfeff || str.charCodeAt(0) === 0xffef) {
+    str = str.substring(1);
+  }
+
+  return str;
+}
+
+/**
+ *
+ * @ignore
+ * Writes a single property for an entry or complex type.
+ *
+ * @param {object} parentElement         Parent DOM element under which the property should be added.
+ * @param {string} name                  Property name.
+ * @param {object} value                 Property value.
+ * @return {object} The current DOM element.
+ *
+ * {Deprecated}
```

Comment:
Is this really deprecated? Should we be adding it if that's the case?

Reply:
Removed the tag - I think overall the XML utilities being used are old, buggy ones. Recall discussing about using alternative XML parsers with @AlexGhiondea, we aren't taking up that experiment for now

Reply:
Is there anything generally (or obviously) wrong with this utility that might have been missed? @chradek @ramya-rao-a 
It's not too clear why it was marked as deprecated and I assumed it was to do with the XML helper lib used.

Reply:
Marking as resolved as I dont see this code in the latest set of changes in the PR